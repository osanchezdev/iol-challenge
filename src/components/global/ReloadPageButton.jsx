import React from 'react';
import Button from 'react-bootstrap/Button';

const ReloadPageButton = () => {
  const reloadPage = () => window.location.reload();

  return (
    <Button variant="secondary" size="lg" onClick={() => reloadPage()}>
      Reload Page
    </Button>
  );
};

export default ReloadPageButton;
