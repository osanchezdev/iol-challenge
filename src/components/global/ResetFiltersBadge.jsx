import React from 'react';
import {Link} from 'react-router-dom';
import {Badge} from 'react-bootstrap';

const ResetFiltersBadge = () => {
  return (
    <Link
      to={{
        pathname: '/',
        search: '',
      }}>
      <Badge pill variant="light">
        Reset filters
      </Badge>
    </Link>
  );
};

export default ResetFiltersBadge;
