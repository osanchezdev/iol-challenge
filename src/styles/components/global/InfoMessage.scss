.info-message--wrapper {
  width: 100%;
  text-align: center;
  & .info-message--emoji {
    font-size: 70px;
  }
  & .info-message--reload {
    cursor: pointer;
  }
}
