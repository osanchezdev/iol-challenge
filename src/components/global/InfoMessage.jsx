import React from 'react';
import {string, bool} from 'prop-types';
import {motion} from 'framer-motion';

import '../../styles/components/global/InfoMessage.scss';
import ResetFiltersBadge from './ResetFiltersBadge';
import ReloadPageButton from './ReloadPageButton';

const InfoMessage = ({emoji, message, badgeLink, reloadButton}) => {
  return (
    <motion.div
      className="info-message--wrapper"
      initial={{scale: 0, opacity: 0}}
      animate={{scale: 1, opacity: 1}}
      transition={{
        type: 'spring',
        stiffness: 260,
        damping: 20,
      }}>
      {emoji && (
        <div className="info-message--emoji">
          <span>{emoji}</span>
        </div>
      )}
      <h5 className="no-result--text">{message}</h5>
      {badgeLink && (
        <div className="info-message--link">
          <ResetFiltersBadge />
        </div>
      )}
      {reloadButton && (
        <div className="info-message--reload">
          <ReloadPageButton />
        </div>
      )}
    </motion.div>
  );
};

InfoMessage.propTypes = {
  emoji: string.isRequired,
  message: string.isRequired,
  badgeLink: bool,
  reloadButton: bool,
};

InfoMessage.defaultProps = {
  emoji: '&#9996;',
  message: 'Hi there!',
};

export default InfoMessage;
