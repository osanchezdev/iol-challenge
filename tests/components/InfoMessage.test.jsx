import * as React from 'react';
import {shallow} from 'enzyme';
import InfoMessage from '../../src/components/global/InfoMessage';
import ResetFiltersBadge from '../../src/components/global/ResetFiltersBadge';
import ReloadPageButton from '../../src/components/global/ReloadPageButton';

describe('Tests in <InfoMessage />', () => {
  test('<InfoMessage /> must be rendered correctly', () => {
    const wrapper = shallow(<InfoMessage />);

    expect(wrapper).toMatchSnapshot();
  });

  test('<InfoMessage /> must show default message', () => {
    const defaultMessage = 'Hi there!';
    const wrapper = shallow(<InfoMessage />);
    const message = wrapper.find('h5').text();
    expect(message).toBe(defaultMessage);
  });

  test('<InfoMessage /> must show default emoji', () => {
    const defaultEmoji = '&#9996;';
    const wrapper = shallow(<InfoMessage />);
    const message = wrapper.find('span').text();

    expect(message).toBe(defaultEmoji);
  });

  test('<InfoMessage /> must show resetBadge with badgeLink prop', () => {
    const wrapper = shallow(<InfoMessage badgeLink />);
    const badgeLinkWrapper = wrapper.find(ResetFiltersBadge);

    expect(badgeLinkWrapper).toHaveLength(1);
  });

  test('<InfoMessage /> must show reloadPageButton with reloadButton prop', () => {
    const wrapper = shallow(<InfoMessage reloadButton />);
    const reloadButtonWrapper = wrapper.find(ReloadPageButton);

    expect(reloadButtonWrapper).toHaveLength(1);
  });
});
