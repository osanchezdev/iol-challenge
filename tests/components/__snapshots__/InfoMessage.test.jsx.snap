// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <InfoMessage /> <InfoMessage /> must be rendered correctly 1`] = `
<ForwardRef(MotionComponent)
  animate={
    Object {
      "opacity": 1,
      "scale": 1,
    }
  }
  className="info-message--wrapper"
  initial={
    Object {
      "opacity": 0,
      "scale": 0,
    }
  }
  transition={
    Object {
      "damping": 20,
      "stiffness": 260,
      "type": "spring",
    }
  }
>
  <div
    className="info-message--emoji"
  >
    <span>
      &#9996;
    </span>
  </div>
  <h5
    className="no-result--text"
  >
    Hi there!
  </h5>
</ForwardRef(MotionComponent)>
`;
